### PR TITLE
minor rework of SendMoney in rpcwallet

### PR DESCRIPTION
**Pull request by Diapolo**
- rework the function to not log errors but use throw JSONRPCError
- remove a check for IsLocked() that is done in sendtoaddress and
  sendfrom RPC calls already
- cache GetBalance() return value, because it's possibly used twice

### CODE REVIEW

File: src/rpcwallet.cpp [hunk 1]
```diff
@@ -310,33 +311,27 @@ Value getaddressesbyaccount(const Array& params, bool fHelp)
...
         throw JSONRPCError(RPC_WALLET_INSUFFICIENT_FUNDS, "Insufficient funds");
 
-    string strError;
-    if (pwalletMain->IsLocked())
-    {
-        strError = "Error: Wallet locked, unable to create transaction!";
-        LogPrintf("SendMoney() : %s", strError);
-        throw JSONRPCError(RPC_WALLET_ERROR, strError);
-    }
-
```
Comment (jonasschnelli) [inline src/rpcwallet.cpp]: Why did you remove this part?

Comment (Diapolo) [inline src/rpcwallet.cpp]: See commit msg ``remove a check for IsLocked() that is done in sendtoaddress and sendfrom RPC calls already``.

File: src/rpcwallet.cpp [hunk 2]
```diff
@@ -310,33 +311,27 @@ Value getaddressesbyaccount(const Array& params, bool fHelp)
 
 void SendMoney(const CTxDestination &address, CAmount nValue, CWalletTx& wtxNew)
```
Comment (laanwj) [inline src/rpcwallet.cpp]: Let's make the function static too

**Comment by laanwj** [comment]
looks good to me, utACK

File: src/rpcwallet.cpp (hunk 2 above)
Comment (Diapolo) [inline src/rpcwallet.cpp]: Done, should be ready now!

**Comment by jonasschnelli** [comment]
ACK.
Passes all tests (`sendtoaddress` is heavily covered by tests and uses `SendMoney()`).

**Comment by fanquake** [comment]
utACK

File: src/rpcwallet.cpp (hunk 1 above)
Comment (TheBlueMatt) [inline src/rpcwallet.cpp]: I'd generally prefer to keep double-checks like this.

**Comment by fanquake** [comment]
Needs rebase

**Comment by Diapolo** [comment]
@fanquake Rebased

**Comment by Diapolo** [comment]
@laanwj What's the state of this?

**Comment by sipa** [comment]
utACK

**Comment by Diapolo** [comment]
Ping
